harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -124,7 +119,8 @@ class Epoch(TrainUnit):
     Epoch step type (e.g. Epoch(1) defines 1 epoch)
     """
 
-    pass
+    def __str__(self):
+        return f"Epoch(value={self.value})"
```

Comment:
nit: `f"Epoch({self.value})"`; better to render it as the user is going to type it out.

Same with Batch.